src/words/tweet_tokenizer.jl

```diff
@@ -0,0 +1,260 @@
+                    WORDS_WITHOUT_APOSTROPHE_DASHES.pattern
+                    ELLIPSIS_DOTS.pattern
+                    r"(?:\S)".pattern
+                    ], "|")
+                    * ")"
+                    )
+
+
+# WORD_REGEX performs poorly on these patterns:
+const HANG_REGEX = r"""([^a-zA-Z0-9])\1{3,}"""
+
+# Regex for replacing HTML_Entities
+const HTML_ENTITIES_REGEX = r"""&(#?(x?))([^&;\s]+);"""
+
+const HANDLES_REGEX = r"""(?x)
+                (?<![A-Za-z0-9_!@#\$%&*])@(([A-Za-z0-9_]){20}(?!@))
+                |
+                (?<![A-Za-z0-9_!@#\$%&*])@(([A-Za-z0-9_]){1,19})(?![A-Za-z0-9_]*@)
+                """
+
+
+"""
+    replace_html_entities(input_text::AbstractString,
+                    remove_illegal=true) => (entities_replaced_text::AbstractString)
+
+Removes entities from text by converting them to their corresponding unicode character.
+`input_text::AbstractString` The string on which HTML entities need to be replaced
+`remove_illegal::Bool` If `true`, entities that can't be converted are
+removed. Otherwise, entities that can't be converted are kept "as
+is".
+Returns `entities_replaced_text::AbstractString`
+"""
+function replace_html_entities(input_text::AbstractString; remove_illegal=true)
+
+    function convert_entity(matched_text)
+        # HTML entity can be named or encoded in Decimal/Hex form
+        # - Named_entity : "&Delta;" => "Δ",
+        # - Decimal : "&#916;" => "Δ",
+        # - Hex : ""&#x394;" => "Δ",
+        #
+        # However for bytes (hex) 80-9f are interpreted in Windows-1252
+        is_numeric_encoded, is_hex_encoded, entity_text = match(HTML_ENTITIES_REGEX,
+                                                matched_text).captures
+        number = -1
+
+        if isempty(is_numeric_encoded)
+            return lookupname(HTML_Entities.default, entity_text)
+        else
+            if isempty(is_hex_encoded)
+                is_numeric = all(isdigit, entity_text)
+                if is_numeric
+                    number = parse(Int, entity_text, base=10)
+                end
+            else
+                base_16_letters = ('a', 'b', 'c', 'd', 'e', 'f')
+                is_base_16 = all(entity_text) do i
+                    isdigit(i) || i in base_16_letters
+                end
+                if is_base_16
+                    number = parse(Int, entity_text, base=16)
+                end
+            end
+
+            # Numeric character references in the 80-9F range are typically
+            # interpreted by browsers as representing the characters mapped
+            # to bytes 80-9F in the Windows-1252 encoding. For more info
+            # see: https://en.wikipedia.org/wiki/ISO/IEC_8859-1#Similar_character_sets
+
+            if number >= 0
+                if 0x80 <= number <= 0x9F
+                    if number ∉ (129, 141, 143, 144, 157)
+                        return decode([UInt8(number)], "WINDOWS-1252")
```

Comment:
This is only 26 characters from "WINDOWS-1252"
What if we just did

```suggestion
                        windows_1252_chars  = ['€', '\\u81', '‚', 'ƒ', '„', '…', '†', '‡', 'ˆ', '‰', 'Š', '‹', 'Œ', '\\u8d', 'Ž', '\\u8f', '\\u90', '‘', '’
', '“', '”', '•', '–', '—', '˜', '™', 'š', '›', 'œ', '\\u9d', 'ž', 'Ÿ')
                       char_ind = number - 0x80 + 1
                       return windows_1252_chars[char_ind]
                        return decode([UInt8(number)], "WINDOWS-1252")
```
(Do check I have this right,  using  tests)

Then we could drop the dependency on StringEncodings.

Reply:
I was looking at how nltk's `tweet_tokenize` handles that the characters "\u81", "\u8d", and other characters for which Windows - 1252 encoding doesn't exist.
I observed that for such case, their code simply removes this character. However, the current code can handle such entities and keep them as-is. Should it be kept the same way, or be matched exactly to the nltk's implementation?

Reply:
@oxinabox Can you give your views this?